radicle-httpd/src/api/v1/projects.rs

```diff
@@ -0,0 +1,392 @@
+        .remotes()?
+        .filter_map(|r| r.map(|r| r.1).ok())
+        .collect::<Vec<_>>();
+
+    Ok::<_, Error>(Json(remotes))
+}
+
+/// Get project remote.
+/// `GET /projects/:project/remotes/:peer`
+async fn remote_handler(
+    Extension(ctx): Extension<Context>,
+    Path((project, node_id)): Path<(Id, NodeId)>,
+) -> impl IntoResponse {
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let remote = repo
+        .remotes()?
+        .filter(|r| {
+            if let Ok((id, _)) = r {
+                node_id == *id
+            } else {
+                false
+            }
+        })
+        .filter_map(|r| r.map(|r| r.1).ok())
+        .collect::<Vec<_>>();
+
+    Ok::<_, Error>(Json(remote))
+}
+
+/// Get project source file.
+/// `GET /projects/:project/blob/:sha/*path`
+async fn blob_handler(
+    Extension(ctx): Extension<Context>,
+    Path((project, sha, path)): Path<(Id, Oid, String)>,
+) -> impl IntoResponse {
+    let path = path.strip_prefix('/').ok_or(Error::NotFound)?;
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let blob = radicle_surf::blob::blob(&repo.raw().into(), Some(Sha { sha }), path)?;
+
+    Ok::<_, Error>(Json(blob))
+}
+
+/// Get project readme.
+/// `GET /projects/:project/readme/:sha`
+async fn readme_handler(
+    Extension(ctx): Extension<Context>,
+    Path((project, sha)): Path<(Id, Oid)>,
+) -> impl IntoResponse {
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let paths = &[
+        "README",
+        "README.md",
+        "README.markdown",
+        "README.txt",
+        "README.rst",
+        "Readme.md",
+    ];
+
+    for path in paths {
+        if let Ok(blob) = radicle_surf::blob::blob(&repo.raw().into(), Some(Sha { sha }), path) {
+            return Ok::<_, Error>(Json(blob));
+        }
+    }
+
+    Err(radicle_surf::object::Error::PathNotFound(
+        radicle_surf::file_system::Path::try_from("README").unwrap(),
+    ))?
+}
+
+/// Get project patches list.
+/// `GET /projects/:project/patches`
+async fn patches_handler(
+    Extension(ctx): Extension<Context>,
+    Path(project): Path<Id>,
+) -> impl IntoResponse {
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let cobs = Store::open(ctx.profile.public_key, &repo)?;
+    let patches = cobs.patches().all()?;
+
+    Ok::<_, Error>(Json(patches))
+}
+
+/// Get project patch.
+/// `GET /projects/:project/patches/:id`
+async fn patch_handler(
+    Extension(ctx): Extension<Context>,
+    Path((project, patch_id)): Path<(Id, ObjectId)>,
+) -> impl IntoResponse {
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let cobs = Store::open(ctx.profile.public_key, &repo)?;
+    let patch = cobs.patches().get(&patch_id)?;
+
+    Ok::<_, Error>(Json(patch))
+}
+
+/// Get project issues list.
+/// `GET /projects/:project/issues`
```

Comment:
Same questions for issues

Reply:
[added](https://github.com/radicle-dev/heartwood/pull/113/files#diff-85a73839d407aa508b9e78b3756b0e9b6e8d1087a0cafac6a8325a8cf427b65eR322)